### PR TITLE
[DEV] - Adds ability to add extra storage for specific address

### PR DESCRIPTION
**Pull request by nadeem-fileverse**
Ticket - https://linear.app/fileverse/issue/ENG-103

### CODE REVIEW

File: src/utils/storageUtils/index.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const EXTRA_STORAGE_ADDRESS_MAP = {
+  "0x5f23Cc6F34640396b0B7023292bee7dc4C2b137B": 100000000, // 100MB in bytes
```
Comment (vijaykrishnavanshi) [inline src/utils/storageUtils/index.js:2]: Can you store and pull this info from database?

Comment (nadeem-fileverse) [reply src/utils/storageUtils/index.js]: Done @vijaykrishnavanshi

File: src/interface/middleware/canUpload.js [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
+const getStorageUse = require("../../domain/limit/getStorageUse");
+const ErrorHandler = require("../../infra/errorHandler");
 
 async function canUpload(req, res, next) {
   const invokerAddress = req.invokerAddress;
   const contractAddress = req.contractAddress;
   if (req.isAuthenticated) {
     const limit = await getStorageUse({ contractAddress });
-    if (limit.storageUse < limit.storageLimit) {
+    const totalAllowedStorage = limit.storageLimit + limit.extraStorage;
```
Comment (vijaykrishnavanshi) [inline src/interface/middleware/canUpload.js:9]: We should handle the case if this value is null or undefined.

**Review by vijaykrishnavanshi — CHANGES_REQUESTED** [review]

File: src/domain/limit/getStorageUse.js [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
+
   return {
     contractAddress,
-    storageLimit: limit && limit.storageLimit || config.DEFAULT_STORAGE_LIMIT,
-    storageUse: limit && limit.storageUse || 0,
-    unit: limit && limit.unit || 'bytes',
+    storageLimit: limit.storageLimit || config.DEFAULT_STORAGE_LIMIT,
+    storageUse: (limit && limit.storageUse) || 0,
+    unit: (limit && limit.unit) || "bytes",
+    extraStorage: limit.extraStorage || 0,
```
Comment (nadeem-fileverse) [inline src/domain/limit/getStorageUse.js:12]: Its handled here @vijaykrishnavanshi

File: src/interface/middleware/canUpload.js (hunk 2 above)
Comment (nadeem-fileverse) [reply src/interface/middleware/canUpload.js:9]: Null values are handled in `getStorageUse` function

**Pull request merged**